packages/react-devtools-shared/src/devtools/views/Components/TreeContext.js

```diff
@@ -372,6 +380,71 @@ function reduceTreeState(store: Store, state: State, action: Action): State {
           }
         }
         break;
+      case 'SELECT_PREVIOUS_ELEMENT_WITH_ERROR_OR_WARNING_IN_TREE': {
+        if (store.errorsAndWarnings.size === 0) {
+          return state;
+        }
+        // TODO (inline errors) The element indices are incorrect if the element is hidden in a collapsed tree.
+        // Shouldn't hidden elements excluded from computing a valid index?
```

Comment:
Filtered elements should be excluded. Hidden elements should auto-expand to be visible, exactly like the Search UX works. :)

Reply:
Ah, so that's the distinction. That is definitely the better UX :+1:

Reply:
And we probably want to re-use the icons from the Search UX as well?

Reply:
Just use whatever's easiest for now, icon/style wise. If you don't mind that is. I'll want to go over the UI and play around with it anyway 😁

Reply:
I'm a bit lost what the source of truth is in the state. It's not obvious if the index controls the ID or vice versa. Tried setting the ID instead but then it's not scrolling down. Could you describe what the fields in the state describe and which fields control the selected index?